molecule/default/molecule.yml

```diff
@@ -0,0 +1,40 @@
+---
+driver:
+  name: ${CONTAINER_BIN:-docker}
+platforms:
+  - name: qdrtest
+    image: docker.io/tripleomaster/centos-binary-qdrouterd:current-tripleo-rdo
+    dockerfile: ../common/Dockerfile.j2
+provisioner:
+  name: ansible
+  log: true
+  playbooks:
+    prepare: ../common/prepare.yml
```

Comment:
AFAICT, the issue is that the CONTAINER_BIN env var is not available to molecule, so using container_bin: ${CONTAINER_BIN:-docker} doesn't work in line 18.

I suggest the following change here:
env:
  CONTAINER_BIN: ${CONTAINER_BIN:-docker}

and change molecule/common/verify.yml to include
vars:
    container_bin: "{{ lookup('env', 'CONTAINER_BIN') }}"

Reply:
Actually, this just makes it so it doesn't ignore $CONTAINER_BIN, and actually sets a value (i.e. defaults to docker) in the molecule container/test environment

Reply:
It works fine as of the last commit. You were right the first time and once I tracked it down, it was obvious.

Reply:
> One thing that's bothering me is the amount of molecule code duplication between this and the qdr-config-ansible-role, though I noticed that you did essentially the same thing with collectd. I wonder if we could make the lower level tests a bit more configurable and then somehow call them from the higher level ones instead of duplicating them.

I tried to avoid duplicating the tests, but it's especially obvious with collectd_config/test_network and tripleo_collectd/collectd_connection. For the latter, the vars were set through tripleo_collectd and propagated to collectd_config, but I didn't see a way to avoid this duplication, as I needed to verify that collectd was configured correctly using tripleo.

Reply:
:D Guess I should have left it at that besides doing some hasty testing yesterday